cpp/include/cudf/strings/string.cuh

```diff
@@ -52,6 +52,42 @@ __device__ bool is_integer(string_view const& d_str)
            thrust::seq, begin, end, [] __device__(auto chr) { return chr >= '0' && chr <= '9'; });
 }
 
+/**
+ * @brief Returns `true` if all characters in the string
+ * are valid for conversion to an integer.
+ * 
+ * It's similar to `is_integer`, The difference is that it would
+ * take an allow_decimal boolean value to say if we want to check 
+ * that the format is [+-]?[0-9]+ like is_integer or if we should 
+ * check that it matches [+-]?[0-9]+(.[0-9]+)? similar to is_float 
+ * but without some of the special cases for float (E, Inf, -Inf, NaN)
+ * 
+ * @param d_str String to check.
+ * @param allow_decimal Decimal format or not
+ * @return true if string has valid integer characters
+ */
+__device__ bool is_valid_integer(string_view const& d_str ,bool allow_decimal)
+{
+  bool decimal_found  = false;
+  if (!allow_decimal) return is_integer(d_str);
```

Comment:
nit: I think we are not going to be able to use `is_integer` here in the long term.  Especially if we want to support range checking. I think the spark code that does this is a decent starting point.

https://github.com/apache/spark/blob/7b06acc28b5c37da6c48bc44c3d921309d4ad3a8/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L1120-L1194

You would want to pass in a `min_value` that corresponds to the type you are checking.  You also would never return the resulting value in any way, just the true/false.